### PR TITLE
Make private keys deterministic in tests

### PR DESCRIPTION
**Pull request by pavel-kokolemin**
Make private keys generation deterministic in tests (from the @TheQuantumPhysicist request).
I switched to using seeded RNG where it was available.
There are some places (not many) where it's not available.
I did not change keys generations in that parts.

### CODE REVIEW

**Comment by TheQuantumPhysicist** [comment]
Closes #541

**Comment by TheQuantumPhysicist** [comment]
> There are some places (not many) where it's not available.
> I did not change keys generations in that parts.

If it's not available because the test doesn't have rng, please add it, just like the other tests. I, personally, had to add many when I supported this for block time.

File: test-utils/src/nft_utils.rs [hunk 1]
```diff
@@ -23,14 +23,12 @@ use common::{
...
-    key::{KeyKind, PrivateKey},
-    random::Rng,
-};
+use crypto::key::{KeyKind, PrivateKey};
+use crypto::random::{CryptoRng, Rng};
 use serialization::extras::non_empty_vec::DataOrNoVec;
 
-pub fn random_creator() -> TokenCreator {
-    let (_, public_key) = PrivateKey::new(KeyKind::RistrettoSchnorr);
+pub fn random_creator(rng: &mut (impl Rng + CryptoRng)) -> TokenCreator {
```
Comment (TheQuantumPhysicist) [inline test-utils/src/nft_utils.rs:30]: Thank you for changing this!

File: crypto/src/key/mod.rs [hunk 2]
```diff
@@ -64,10 +65,16 @@ impl From<RistrettoSignatureError> for SignatureError {
 
 impl PrivateKey {
     pub fn new(key_kind: KeyKind) -> (PrivateKey, PublicKey) {
```
Comment (TheQuantumPhysicist) [inline crypto/src/key/mod.rs:67]: Can you rename this to `new_from_entropy`?

**Review by TheQuantumPhysicist — APPROVED** [review]
Looking fine. Please try to eliminate all the entropy-based derivations of the private key you can.

**Review by stanislav-tkach — APPROVED** [review]

File: crypto/src/key/mod.rs (hunk 2 above)
Comment (pavel-kokolemin) [reply crypto/src/key/mod.rs]: Updated, thanks.

**Comment by pavel-kokolemin** [comment]
> Looking fine. Please try to eliminate all the entropy-based derivations of the private key you can.

Update in all remaining tests.

**Pull request merged**